File: Frontend/trackr-administrador/src/app/shared/models/catalogo/dominio.ts
import { DominioDetalle } from "@models/catalogo/dominio-detalle";

export class Dominio {
    public idDominio: number;
    public clave: string;
    public nombre: string;
    public descripcion: string;
    public tipoDato: string;
    public tipoCampo: string;
    public longitudMinima: number;
    public longitudMaxima: number;
    public valorMinimo: number;
    public valorMaximo: number;
    public fechaMinima: Date;
    public fechaMaxima: Date;
    public permiteFueraDeRango: boolean;
    public unidadMedida: string;

    public dominioDetalle: DominioDetalle[];

    constructor() {}
}

File: Backend/TrackrAPI/TrackrAPI/Models/Dominio.cs
﻿using System;
using System.Collections.Generic;

namespace TrackrAPI.Models
{
    public partial class Dominio
    {
        public Dominio()
        {
            DominioDetalle = new HashSet<DominioDetalle>();
            DominioHospital = new HashSet<DominioHospital>();
            SeccionCampo = new HashSet<SeccionCampo>();
        }

        public int IdDominio { get; set; }
        public string Nombre { get; set; } = null!;
        public string Descripcion { get; set; } = null!;
        public string TipoDato { get; set; } = null!;
        public string TipoCampo { get; set; } = null!;
        public int? LongitudMinima { get; set; }
        public int? LongitudMaxima { get; set; }
        public decimal? ValorMinimo { get; set; }
        public decimal? ValorMaximo { get; set; }
        public DateTime? FechaMinima { get; set; }
        public DateTime? FechaMaxima { get; set; }
        public bool? PermiteFueraDeRango { get; set; }
        public string? UnidadMedida { get; set; }
        public int? IdHospital { get; set; }

        public virtual Hospital? IdHospitalNavigation { get; set; }
        public virtual ICollection<DominioDetalle> DominioDetalle { get; set; }
        public virtual ICollection<DominioHospital> DominioHospital { get; set; }
        public virtual ICollection<SeccionCampo> SeccionCampo { get; set; }
    }
}
